tests/components/dynalite/test_bridge.py

```diff
@@ -1,136 +1,94 @@
-    host = "1.2.3.4"
-    entry.data = {"host": host}
-    hass.data = {DOMAIN: {DATA_CONFIGS: {host: {}}}}
-    dyn_bridge = DynaliteBridge(hass, entry)
-
     device1 = Mock()
     device1.category = "light"
     device2 = Mock()
     device2.category = "switch"
     reg_func = Mock()
-    dyn_bridge.register_add_entities(reg_func)
-    dyn_bridge.add_devices([device1, device2])
+    dyn_bridge.register_add_devices(reg_func)
+    dyn_bridge.add_devices_when_registered([device1, device2])
     reg_func.assert_called_once()
-    assert reg_func.mock_calls[0][1][0][0].device is device1
-
-
-async def test_update_device():
-    """Test the update_device callback."""
-    hass = Mock()
-    entry = Mock()
-    host = "1.2.3.4"
-    entry.data = {"host": host}
-    hass.data = {DOMAIN: {DATA_CONFIGS: {host: {}}}}
-    dyn_bridge = DynaliteBridge(hass, entry)
-    with patch.object(dyn_bridge, "dynalite_devices") as devices_mock:
-        # Single device update
-        device1 = Mock()
-        device1.unique_id = "testing1"
-        device2 = Mock()
-        device2.unique_id = "testing2"
-        dyn_bridge.all_entities = {
-            device1.unique_id: device1,
-            device2.unique_id: device2,
-        }
-        dyn_bridge.update_device(device1)
-        device1.try_schedule_ha.assert_called_once()
-        device2.try_schedule_ha.assert_not_called()
-        # connected to network - all devices update
-        devices_mock.available = True
-        dyn_bridge.update_device(CONF_ALL)
-        assert device1.try_schedule_ha.call_count == 2
-        device2.try_schedule_ha.assert_called_once()
-        # disconnected from network - all devices update
-        devices_mock.available = False
-        dyn_bridge.update_device(CONF_ALL)
-        assert device1.try_schedule_ha.call_count == 3
-        assert device2.try_schedule_ha.call_count == 2
-
-
-async def test_async_reset():
-    """Test async_reset."""
-    hass = Mock()
-    hass.config_entries.async_forward_entry_unload = Mock(
-        return_value=mock_coro(Mock())
-    )
-    entry = Mock()
-    host = "1.2.3.4"
-    entry.data = {"host": host}
-    hass.data = {DOMAIN: {DATA_CONFIGS: {host: {}}}}
-    dyn_bridge = DynaliteBridge(hass, entry)
-    await dyn_bridge.async_reset()
-    hass.config_entries.async_forward_entry_unload.assert_called_once()
-    assert hass.config_entries.async_forward_entry_unload.mock_calls[0] == call(
-        entry, "light"
-    )
+    assert reg_func.mock_calls[0][1][0][0] is device1
+
+
+async def test_try_connection(dyn_bridge):
+    """Test that try connection works."""
+    # so it doesn't complain asyncio is not used
+    asyncio.get_running_loop()
+    # successful
+    with patch.object(dyn_bridge.dynalite_devices, "connected", True):
+        assert await dyn_bridge.try_connection()
+    # unsuccessful
+    with patch.object(dyn_bridge.dynalite_devices, "connected", False):
+        with patch("asyncio.sleep", return_value=True):
```

Comment:
Don't patch asyncio. Patch the sleep time and set it to `0`.

To make this easier, extract the sleep time variabel to a module level constant.

Reply:
done